src/lib/statistics/cardinality_estimator.cpp

```diff
@@ -0,0 +1,858 @@
+        const auto right_histogram = right_input_column_statistics->histogram;
+        if (!left_histogram || !right_histogram) {
+          // TODO(anyone) Can only use histograms to estimate column-to-column scans right now
+          selectivity = 1.0f;
+          return;
+        }
+
+        const auto bin_adjusted_left_histogram = left_histogram->split_at_bin_bounds(right_histogram->bin_bounds());
+        if (!bin_adjusted_left_histogram) {
+          selectivity = 1.0f;
+          return;
+        }
+
+        const auto bin_adjusted_right_histogram = right_histogram->split_at_bin_bounds(left_histogram->bin_bounds());
+        if (!bin_adjusted_right_histogram) {
+          selectivity = 1.0f;
+          return;
+        }
+
+        const auto column_vs_column_histogram = estimate_column_vs_column_equi_scan_with_histograms(
+            *bin_adjusted_left_histogram, *bin_adjusted_right_histogram);
+        if (!column_vs_column_histogram) {
+          selectivity = 0.0f;
+          return;
+        }
+
+        const auto cardinality = column_vs_column_histogram->total_count();
+        selectivity = input_table_statistics->row_count == 0 ? 0.0f : cardinality / input_table_statistics->row_count;
+
+        /**
+         * Write out the ColumnStatistics of the scanned columns
+         */
+        const auto column_statistics = std::make_shared<ColumnStatistics<ColumnDataType>>();
+        column_statistics->histogram = column_vs_column_histogram;
+        output_column_statistics[left_column_id] = column_statistics;
+        output_column_statistics[*right_column_id] = column_statistics;
+
+      } else if (predicate.value.type() == typeid(ParameterID)) {
+        /**
+         * Estimate ColumnVsPlaceholder
+         */
+
+        switch (predicate.predicate_condition) {
+          case PredicateCondition::Equals: {
+            const auto total_distinct_count =
+                std::max(scan_statistics_object->total_distinct_count(), HistogramCountType{1.0f});
+            selectivity = total_distinct_count > 0 ? 1.0f / total_distinct_count : 0.0f;
+          } break;
+
+          case PredicateCondition::NotEquals: {
+            const auto total_distinct_count =
+                std::max(scan_statistics_object->total_distinct_count(), HistogramCountType{1.0f});
+            selectivity = total_distinct_count > 0 ? (total_distinct_count - 1.0f) / total_distinct_count : 0.0f;
+          } break;
+
+          case PredicateCondition::LessThan:
+          case PredicateCondition::LessThanEquals:
+          case PredicateCondition::GreaterThan:
+          case PredicateCondition::GreaterThanEquals:
+          case PredicateCondition::Between:
+          case PredicateCondition::In:
+          case PredicateCondition::NotIn:
+          case PredicateCondition::Like:
+          case PredicateCondition::NotLike:
+            // Lacking better options, assume a "magic" selectivity for >, >=, <, <=, ....
+            selectivity = 0.5f;
```

Comment:
Sometimes the "dunno selectivity" is 1, here it is .5. Make it a constant?

Reply:
But if sometimes a selectivity of 0.5f just is a better guess and sometimes a 1 is, what value would that constant have? Or one constant for each magic number? I don't think that would improve it in this case.

Reply:
If they were deliberately chosen, maybe add a one-line comment on how we ended up with this number?

Reply:
This is awkward, really. I have now arrived at this comment:
```
            // Lacking better options, assume a "magic" selectivity for >, >=, <, <=, ... Any number would be equally
            // right and wrong here. In some examples, this seemed like a good guess ¯\_(ツ)_/¯
```

It doesn't get much better imo. For some magic numbers the justification is literally that: it seemed to be a good choice in an example I looked at.